waku/v2/utils/peers.nim

```diff
@@ -112,7 +112,49 @@ func getTransportProtocol(typedR: TypedRecord): Option[IpTransportProtocol] =
 
 ## Parses a fully qualified peer multiaddr, in the
 ## format `(ip4|ip6)/tcp/p2p`, into dialable PeerInfo
-proc parseRemotePeerInfo*(address: string): RemotePeerInfo {.raises: [Defect, ValueError, LPError].}=
+proc parsePeerInfo*(address: string):
+                    Result[RemotePeerInfo, string] =
+  var
+    nwPart, tcpPart, p2pPart, wsPart, wssPart: MultiAddress
+
+  try:
+    let multiAddr = MultiAddress.init(address).tryGet()
```

Comment:
Note that the `tryGet()` method is a `stew/results` package procedure/template that converts an error into an exception (similar to `unwrap` in Rust).

Here the correct thing would be:

- Either bubble up the error using the `?` operator:
```suggestion
    let multiAddr = ? MultiAddress.init(address)
```
- Or bubble up the error with a wrapped error info via `?` operator and the `mapErr` method:

```suggestion
    let multiAddr = ? MultiAddress.init(address).mapErr(proc(err: string): string = "invalid multiaddress format: " & err)
```

The option depends on what error message do you want to provide.

Reply:
I like more the second option, which gives more info.